BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -504,45 +507,86 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
         // Allow a certificate if its status was evaluated successfully and the result is that it should be trusted
         BOOL shouldTrustServer = (status == errSecSuccess && (trustResult == kSecTrustResultProceed || trustResult == kSecTrustResultUnspecified));
 
-        [self completeServerTrustAuthenticationChallenge:challenge shouldTrust:shouldTrustServer];
+        if (shouldTrustServer) {
+            SecTrustRef serverTrust = [[challenge protectionSpace] serverTrust];
+            NSURLCredential *serverTrustCredential = [NSURLCredential credentialForTrust:serverTrust];
+            if (completionHandler) {
+                completionHandler(NSURLSessionAuthChallengeUseCredential, serverTrustCredential);
+                //FIXME: Or should we use NSURLSessionAuthChallengePerformDefaultHandling
+            } else {
+                [[challenge sender] useCredential:serverTrustCredential
```

Comment:
could we test assuming that completionHandler is nil, and just call this directly to see if session task can still finish successfully?